lib/cisco_node_utils/interface.rb

```diff
@@ -303,6 +303,39 @@ def default_negotiate_auto
       config_get_default('interface', negotiate_auto_lookup_string)
     end
 
+    def port_channel
+      config_get('interface', 'port_channel', @name)
+    end
+
+    def port_channel=(val)
+      case @name
+      when /Ethernet/i
+        unless val.empty?
```

Comment:
Please go back to the drawing board on this setter. It is entirely too complex for what it needs to do, or else this needs a lot of comments to justify what's happening.
This should simply set the channel_group ID and optionally 'mode' (if it's in the scope of this project).
Why do you need to use port_channel_destroy?
I think we should probably have a design session to discuss these changes.